pkg/sql/opt/xform/optimizer.go

```diff
@@ -720,6 +720,15 @@ func (o *Optimizer) optimizeRootWithProps() {
 			if o.appliedRule != nil {
 				o.appliedRule(opt.PruneRootCols, nil, root)
 			}
+			// We may have pruned a column that appears in the required ordering.
+			rootCols := root.Relational().OutputCols
+			if !rootProps.Ordering.SubsetOfCols(rootCols) {
```

Comment:
I think this logic should be moved up and incorporated into the logic before the `SetRoot` call above. Create the `root` via a call to `PruneCols` and then `rootProps` from that, and pass them to `SetRoot`.

Reply:
Done, kind of. I still have two calls to SetRoot to avoid having to re-intern the props when they don't change.